pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,146 @@
+    echo "Running container image to insert the oot kernel module $KERNEL_MODULE"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -d /opt $KERNEL_MODULE
+    echo "OOT kernel module $KERNEL_MODULE is inserted"
+else
+   echo "Image $IMAGE:$TAG is not present in local registry, will try after reboot"
+fi
+`
+const pullKernelModuleScriptTemplate = `#!/bin/bash
+if podman image list | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry.Nothing to do"
+else
+    echo "Image $IMAGE:$TAG not found in the local registry, pulling"
+    podman pull --authfile /var/lib/kubelet/config.json $IMAGE:$TAG
+    if [ $? -eq 0 ]; then
+        echo "Image $IMAGE:$TAG has been successfully pulled, rebooting.."
+        reboot
+    else
+        echo "Failed to pull image $IMAGE:$TAG"
+    fi
+fi
+`
+
+const mcoTemplate = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: $MCO_MACHINE_CONFIG_POOL_REF
+  name: $MCO_NAME
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+        - contents: |
+            [Unit]
+            Description=Replace in-tree kernel module with oot kernel module
+            Before=network-pre.target
+            Wants=network-pre.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            TimeoutSec=10
+            ExecStartPre=ls /usr/local/bin
+            ExecStart=/usr/local/bin/replace-kernel-module.sh
+            PrivateTmp=yes
+            RemainAfterExit=no
+            TimeoutSec=60
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: "replace-kernel-module.service"
+        - contents: |
+            [Unit]
+            Description=Pull oot kernel module image
+            After=network-online.target
+            Wants=network-online.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            ExecStart=/usr/local/bin/pull-kernel-module-image.sh
+            PrivateTmp=yes
+            RemainAfterExit=no
+            TimeoutSec=900
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true
+          name: "pull-kernel-module-image.service"
+        - enabled: false
+          mask: true
+          name: crio-wipe.service
+    storage:
+      files:
+        - path: "/usr/local/bin/replace-kernel-module.sh"
+          mode: 511
+          overwrite: true
+          user:
+            name: "root"
+          contents:
+            source: "data:text/plain;base64,$REPLACE_IN_TREE_DRIVER_CONTENTS"
+        - path: "/usr/local/bin/pull-kernel-module-image.sh"
+          mode: 493
+          overwrite: true
+          user:
+            name: "root"
+          contents:
+            source: "data:text/plain;base64,$PULL_OOT_KERNEL_MODULE_CONTENTS"
+`
```

Comment:
How about we create some data files instead of hardcoding them?

Reply:
and then reading them? I prefer this way, where you can see all the input in one file ( this is the same way we did in AI with all ignition templates.